src/java.base/share/classes/java/net/ProxySelector.java

```diff
@@ -207,12 +208,22 @@ static class StaticProxySelector extends ProxySelector {
 
         @Override
         public void connectFailed(URI uri, SocketAddress sa, IOException e) {
+            if (uri == null || sa == null || e == null) {
+                throw new IllegalArgumentException("Arguments can't be null.");
+            }
             /* ignore */
         }
 
         @Override
-        public synchronized List<Proxy> select(URI uri) {
-            String scheme = uri.getScheme().toLowerCase(Locale.ROOT);
+        public List<Proxy> select(URI uri) {
+            if (uri == null) {
+                throw new IllegalArgumentException("URI can't be null");
+            }
+            String scheme = uri.getScheme();
+            if (scheme == null) {
+                throw new IllegalArgumentException("protocol can't be null");
+            }
+            scheme = scheme.toLowerCase(Locale.ROOT);
             if (scheme.equals("http") || scheme.equals("https")) {
```

Comment:
maybe outside the scope of this PR, but could avoid the `toLowerCase`

```suggestion
            if (scheme.equalsIgnoreCase("http") || scheme.equalsIgnoreCase("https")) {
```

Reply:
Done.